### PR TITLE
set crates versions for v1.0.0 release

### PR DESCRIPTION
**Pull request by plebhash**
in the past, we didn't follow `SemVer 2.0.0` while versioning the SRI crates

as we prepare for the `v1.0.0` release, it is a good idea to set all crates into the same baseline, under the assumption that going forward, all SRI crates versioning will follow `SemVer 2.0.0`, as discussed on https://github.com/stratum-mining/stratum/pull/766

the only exceptions are:
- `noise-sv2` crate, which is starting from `v1.1.0`
- `roles` crates, which are still somewhat immature, so they remain at `v0.1.0`

### CODE REVIEW

File: benches/Cargo.toml [hunk 1]
```diff
@@ -10,9 +10,9 @@ async-channel = "1.4.0"
 v1={path="../protocols/v1", package="sv1_api"}
 serde_json = { version = "1.0.64", default-features = false, features = ["alloc"] }
 iai="0.1"
-mining_sv2 = { path = "../protocols/v2/subprotocols/mining", version = "^0.1.0" }
+mining_sv2 = { path = "../protocols/v2/subprotocols/mining", version = "^1.0.0" }
 roles_logic_sv2 = { path = "../protocols/v2/roles-logic-sv2" }
```
Comment (GitGab19) [inline benches/Cargo.toml:14]: why there's not a version specified here?

File: examples/ping-pong-with-noise/Cargo.lock
Comment (GitGab19) [inline examples/ping-pong-with-noise/Cargo.lock:1]: I would remove Cargo.lock here

File: examples/ping-pong-with-noise/Cargo.toml [hunk 2]
```diff
@@ -15,7 +15,7 @@ async-std="1.8.0"
 bytes = "1.0.1"
 binary_sv2 = { path = "../../protocols/v2/binary-sv2/binary-sv2" }
 codec_sv2 = { path = "../../protocols/v2/codec-sv2", features=["noise_sv2"] }
```
Comment (GitGab19) [inline examples/ping-pong-with-noise/Cargo.toml:17]: Why there's not a version specified here?

File: examples/ping-pong-without-noise/Cargo.lock
Comment (GitGab19) [inline examples/ping-pong-without-noise/Cargo.lock:1]: I would remove it

File: protocols/Cargo.toml
Comment (GitGab19) [inline protocols/Cargo.toml:1]: Didn't we updated the MSRV to 1.75.0? @Fi3

File: benches/Cargo.toml [hunk 3]
```diff
@@ -10,9 +10,9 @@ async-channel = "1.4.0"
 v1={path="../protocols/v1", package="sv1_api"}
```
Comment (GitGab19) [inline benches/Cargo.toml:10]: why there's not a version specified here?

File: protocols/fuzz-tests/Cargo.toml
Comment (GitGab19) [inline protocols/fuzz-tests/Cargo.toml:1]: Shouldn't we specify "1.0.0" for `binary_codec_sv2` and `roles_logic_sv2`?

File: roles/Cargo.toml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name = "stratum_v2_roles"
-version = "0.0.0"
+version = "1.0.0"
 authors = ["The Stratum v2 Developers"]
 edition = "2021"
 rust-version = "1.65.0"
```
Comment (GitGab19) [inline roles/Cargo.toml:5]: Same here: shouldn't be 1.75.0? @Fi3

File: roles/test-utils/mining-device/Cargo.toml
Comment (GitGab19) [inline roles/test-utils/mining-device/Cargo.toml:1]: Version for dependencies are missing

File: utils/message-generator/Cargo.toml [hunk 5]
```diff
@@ -13,7 +13,7 @@ const_sv2 = { path = "../../protocols/v2/const-sv2" }
 load_file = "1.0.1"
 network_helpers = { path = "../../utils/network-helpers", features = ["with_tokio","with_serde"] }
 roles_logic_sv2 = { path = "../../protocols/v2/roles-logic-sv2", features = ["with_serde"] }
```
Comment (GitGab19) [inline utils/message-generator/Cargo.toml:15]: Version missing

File: utils/buffer/fuzz/Cargo.lock
Comment (GitGab19) [inline utils/buffer/fuzz/Cargo.lock:1]: I would remove it

Comment (plebhash) [reply utils/buffer/fuzz/Cargo.lock:1]: ack https://github.com/stratum-mining/stratum/pull/785/commits/c438a5b49b90950025a444363cf3a5ba4bb895aa

File: examples/ping-pong-without-noise/Cargo.lock
Comment (plebhash) [reply examples/ping-pong-without-noise/Cargo.lock:1]: ack https://github.com/stratum-mining/stratum/pull/785/commits/c438a5b49b90950025a444363cf3a5ba4bb895aa

File: examples/ping-pong-with-noise/Cargo.lock
Comment (plebhash) [reply examples/ping-pong-with-noise/Cargo.lock:1]: ack https://github.com/stratum-mining/stratum/pull/785/commits/c438a5b49b90950025a444363cf3a5ba4bb895aa

File: roles/test-utils/mining-device/Cargo.toml
Comment (plebhash) [reply roles/test-utils/mining-device/Cargo.toml:1]: ack https://github.com/stratum-mining/stratum/pull/785/commits/a877dd2960832b743fdbc0bd041f89b10fb252a8

File: utils/message-generator/Cargo.toml (hunk 5 above)
Comment (plebhash) [reply utils/message-generator/Cargo.toml]: ack https://github.com/stratum-mining/stratum/pull/785/commits/a877dd2960832b743fdbc0bd041f89b10fb252a8

File: examples/ping-pong-with-noise/Cargo.toml (hunk 2 above)
Comment (plebhash) [reply examples/ping-pong-with-noise/Cargo.toml]: ack https://github.com/stratum-mining/stratum/pull/785/commits/a877dd2960832b743fdbc0bd041f89b10fb252a8

File: benches/Cargo.toml (hunk 1 above)
Comment (plebhash) [reply benches/Cargo.toml]: basically it was already missing, but I'm adding it here https://github.com/stratum-mining/stratum/pull/785/commits/a877dd2960832b743fdbc0bd041f89b10fb252a8

File: protocols/fuzz-tests/Cargo.lock
Comment (GitGab19) [inline protocols/fuzz-tests/Cargo.lock:1]: I would remove this

File: protocols/Cargo.toml
Comment (pavlenex) [reply protocols/Cargo.toml:1]: We should've, it's even in readme.

File: protocols/fuzz-tests/Cargo.lock
Comment (plebhash) [reply protocols/fuzz-tests/Cargo.lock:1]: ack https://github.com/stratum-mining/stratum/pull/785/commits/b65f079318132e97fade8c7e5b22d1d4a7811616

File: roles/Cargo.toml (hunk 4 above)
Comment (plebhash) [reply roles/Cargo.toml]: ack https://github.com/stratum-mining/stratum/pull/785/commits/5ac08c2a20802c787a914b1cad4613f7748de1c4

File: protocols/Cargo.toml
Comment (plebhash) [reply protocols/Cargo.toml:1]: ack https://github.com/stratum-mining/stratum/pull/785/commits/5ac08c2a20802c787a914b1cad4613f7748de1c4

File: protocols/fuzz-tests/Cargo.lock
Comment (GitGab19) [inline protocols/fuzz-tests/Cargo.lock:1]: I would remove it

**Review by GitGab19 — CHANGES_REQUESTED** [review]
LGTM.
I would just remove `protocols/fuzz-tests/Cargo.lock`

File: protocols/fuzz-tests/Cargo.lock
Comment (plebhash) [reply protocols/fuzz-tests/Cargo.lock:1]: it was already removed on https://github.com/plebhash/stratum/commit/2388fa725547fbb6c96842cb323c568f7a990792

**Review by GitGab19 — APPROVED** [review]

**Review by Fi3 — APPROVED** [review]

**Comment by Fi3** [comment]
LGTM

**Pull request merged**